patch/driver-pca954x-i2c-mux-deselect-on-exit-config-option.patch

```diff
@@ -0,0 +1,62 @@
+Adding PCA9548 I2C MUX deselect on exit config
+
+From: wadelnn <wadelnn@users.noreply.github.com>
+
+---
+ drivers/i2c/muxes/Kconfig           | 7 +++++++
+ drivers/i2c/muxes/i2c-mux-pca954x.c | 7 ++++++-
+ 2 files changed, 13 insertions(+), 1 deletion(-)
+
+diff --git a/drivers/i2c/muxes/Kconfig b/drivers/i2c/muxes/Kconfig
+index f6d313e..83a49d3 100644
+--- a/drivers/i2c/muxes/Kconfig
++++ b/drivers/i2c/muxes/Kconfig
+@@ -48,6 +48,13 @@ config I2C_MUX_PCA954x
+ 	  This driver can also be built as a module.  If so, the module
+ 	  will be called i2c-mux-pca954x.
+ 
++config I2C_MUX_PCA954X_DESELECT_ON_EXIT
++	bool "Enable deselect-on-exit feature for PCA954X devices."
++	depends on I2C_MUX_PCA954x
++	help
++	  If you say yes here you enable the deselect-on-exit feature in
++	  the pca954x i2c driver.
++
+ config I2C_MUX_PINCTRL
+ 	tristate "pinctrl-based I2C multiplexer"
+ 	depends on PINCTRL
+diff --git a/drivers/i2c/muxes/i2c-mux-pca954x.c b/drivers/i2c/muxes/i2c-mux-pca954x.c
+index 9bd4212..07ad443 100644
+--- a/drivers/i2c/muxes/i2c-mux-pca954x.c
++++ b/drivers/i2c/muxes/i2c-mux-pca954x.c
+@@ -188,6 +188,7 @@ static int pca954x_probe(struct i2c_client *client,
+ 	struct gpio_desc *gpio;
+ 	int num, force, class;
+ 	struct pca954x *data;
++    int deselect_on_exit = 0;
+ 	int ret;
+ 
+ 	if (!i2c_check_functionality(adap, I2C_FUNC_SMBUS_BYTE))
+@@ -213,6 +214,10 @@ static int pca954x_probe(struct i2c_client *client,
+ 		return -ENODEV;
+ 	}
+ 
++#ifdef CONFIG_I2C_MUX_PCA954X_DESELECT_ON_EXIT
++    deselect_on_exit = 1;
++#endif
++
+ 	data->type = id->driver_data;
+ 	data->last_chan = 0;		   /* force the first selection */
+ 
+@@ -233,7 +238,7 @@ static int pca954x_probe(struct i2c_client *client,
+ 		data->virt_adaps[num] =
+ 			i2c_add_mux_adapter(adap, &client->dev, client,
+ 				force, num, class, pca954x_select_chan,
+-				(pdata && pdata->modes[num].deselect_on_exit)
++				(pdata && pdata->modes[num].deselect_on_exit) || deselect_on_exit
```

Comment:
The pdata->modes[num].deselect_on_exit is different transaction flag from dev_get_platdata. If other platforms use pca954x module frequently may cause this issue. We want to make sure that the pca954x channel register is set everytime to pca954x_select_chan function in i2c-mux-pca954x.c.